components/net/mime_classifier.rs

```diff
@@ -33,35 +35,107 @@ pub enum NoSniffFlag {
 }
 
 impl MIMEClassifier {
-    //Performs MIME Type Sniffing Algorithm (section 7)
+    //Performs MIME Type Sniffing Algorithm (sections 7 and 8)
     pub fn classify(&self,
+                    context: LoadContext,
                     no_sniff_flag: NoSniffFlag,
                     apache_bug_flag: ApacheBugFlag,
                     supplied_type: &Option<(String, String)>,
-                    data: &[u8]) -> (String, String) {
-        match *supplied_type {
-            None => self.sniff_unknown_type(no_sniff_flag, data),
-            Some(ref supplied_type) => {
-                let &(ref media_type, ref media_subtype) = supplied_type;
-                if MIMEClassifier::is_explicit_unknown(media_type, media_subtype) {
-                    self.sniff_unknown_type(no_sniff_flag, data)
-                } else {
-                    match no_sniff_flag {
-                        NoSniffFlag::ON => supplied_type.clone(),
-                        NoSniffFlag::OFF => match apache_bug_flag {
-                            ApacheBugFlag::ON => self.sniff_text_or_data(data),
-                            ApacheBugFlag::OFF => match MIMEClassifier::get_media_type(media_type,
-                                                                                       media_subtype) {
-                                Some(MediaType::Xml) => None,
-                                Some(MediaType::Html) => self.feeds_classifier.classify(data),
-                                Some(MediaType::Image) => self.image_classifier.classify(data),
-                                Some(MediaType::AudioVideo) => self.audio_video_classifier.classify(data),
-                                None => None
-                            }.unwrap_or(supplied_type.clone())
+                    data: &[u8]) -> Option<(String, String)> {
```

Comment:
Is there some reason you're changing this?  It will substantially simplify downstream code if we can assume MIME sniffing always returns something (even if it's just `application/octet-stream` or something).